usr/share/rear/lib/_framework-setup-and-functions.sh

```diff
@@ -1507,3 +1507,324 @@ function ProgressInfo () {
     : ;
 }
 
+# Sourcing functions to enforce trusted sourcing:
+
+# Specify default TRUSTED_OWNERS:
+#
+# The owner of sbin/rear is always trusted.
+# Usually the owner of sbin/rear is 'root'
+# but if not (e.g. in case of Git checkout on the original system)
+# also 'root' is trusted by default:
+sbin_rear_owner_name="$( stat -L -c %U "$SCRIPT_FILE" )"
+test "$sbin_rear_owner_name" = "root" && TRUSTED_OWNERS=( 'root' ) || TRUSTED_OWNERS=( 'root' "$sbin_rear_owner_name" )
+# 'readonly TRUSTED_OWNERS' happens in sbin/rear
+# after the normal user configuration files (site.conf local.conf rescue.conf) were sourced
+# so that the user can specify what he needs, e.g. for whatever third-party (backup) software.
+
+# Check file owner is trusted
+# see https://relax-and-recover.org/documentation/security-architecture
+function is_trusted_owner () {
+    local file="$1"
+    local owner_name=""
+    local trusted_owner=""
+    # Do not error out in 'stat' when it is neither a regular file nor a link to a regular file
+    # but it is not trusted when it is neither a regular file nor a link to a regular file:
+    test -f "$file" || return 1
+    # Empty TRUSTED_OWNERS means all regular file owners are blindly trusted,
+    # cf https://github.com/rear/rear/pull/3379#issuecomment-2633123866
+    test ${#TRUSTED_OWNERS[@]} -eq 0 && return 0
+    # '-L' forces stat to follow symlinks (and error out if that fails):
+    owner_name="$( stat -L -c %U "$file" )" || Error "is_trusted_owner(): 'stat -L -c %U $file' failed"
+    for trusted_owner in "${TRUSTED_OWNERS[@]}" ; do
+        test "$owner_name" = "$trusted_owner" && return 0
+    done
+    return 1
+}
+
+# Specify default TRUSTED_PATHS:
+#
+# ReaR basic directories cases:
+#
+# (A1)
+# Normal case (i.e. no Git checkout) on the original system:
+# ReaR basic directories when running /usr/sbin/rear on the original system from a normally installed package
+# (i.e. when ReaR's files are installed in '/' so REAR_DIR_PREFIX is empty):
+#   REAR_DIR_PREFIX=
+#   SHARE_DIR=/usr/share/rear
+#   CONFIG_DIR=/etc/rear
+#   VAR_DIR=/var/lib/rear
+#   LOG_DIR=/var/log/rear
+#
+# (A2)
```

Comment:
What is the difference between A1 and A2?

Reply:
The difference is "on the original system" (A1)
versus "in the ReaR recovery system" (A2)
and in both cases REAR_DIR_PREFIX is empty
and the ReaR basic directories are same
on the original system and in the ReaR recovery system.

I listed "on the original system"
versus "in the ReaR recovery system"
sepatated also for case (A no Git checkout)
to have is same for case (A)
as it is then shown for case (B Git checkout)
where REAR_DIR_PREFIX and the ReaR basic directories are
different on the original system and in the ReaR recovery system.

So it is easier to compare case B1 with the matching A1
(both cases show how it looks on the original system)
versus case B2 with the matching A2
(both cases show how it looks in the ReaR recovery system)

Reply:
After re-reading the comments about cases A1 and A2 in
_framework-setup-and-functions.sh I see now that I can
simplify the cases A1 and A2 into one single case A
because I nowhere to handle one of the cases A1 or A2
separated from the other one but always both together
(in contrast to the cases B1 and B2 where I need to
handle B1 separated from B2 via different code).